cores/portduino/main.cpp

```diff
@@ -75,12 +78,18 @@ static error_t parse_opt(int key, char *arg, struct argp_state *state) {
 }
 
 void getMacAddr(uint8_t *dmac) {
-  dmac[0] = 0x80;
-  dmac[1] = 0;
-  dmac[2] = portduinoArguments.hwId >> 24;
-  dmac[3] = portduinoArguments.hwId >> 16;
-  dmac[4] = portduinoArguments.hwId >> 8;
-  dmac[5] = portduinoArguments.hwId & 0xff;
+    std::fstream macIdentity;
+    macIdentity.open("/sys/kernel/debug/bluetooth/hci0/identity", std::ios::in);
+    std::string macLine;
+    getline(macIdentity, macLine);
+    macIdentity.close();
+
```

Comment:
better check if that sys node actually exists. There are linux systems without bluetooth support...